src/lib/yang/data.lua

```diff
@@ -240,7 +240,15 @@ local function range_validator(range, f) return f end
 local function length_validator(range, f) return f end
 local function pattern_validator(range, f) return f end
 local function bit_validator(range, f) return f end
-local function enum_validator(range, f) return f end
+local function enum_validator(enums, f)
+   if not enums or #enums == 0 then return f end
+   return function (val)
```

Comment:
can we instead compute a table of valid values so that the lookup goes faster?  can be a followup